### PR TITLE
TS migration + Refactoring

### PR DESCRIPTION
**Pull request by Hofled**
# TS Migration + Refactoring

Added Typescript to the project and all necessary infrastructural stuff.
Tests are currently broken but they are WIP.
Everything else should function normally, while a lot of bugs were resolved.
Some of the code was migrated to Typescript already, while the majority was kept intact with minor changes (or bug fixes) to allow the migration.

## Things to look out for in this PR:
While going over the code, not a lot of logic has been re-written, but most of the stuff was moved \ renamed, and some bugs were fixed.

Try to focus on the structural changes and bug fixes, and not on the many file removals (as files were converted from `.js` to `.ts`.

A big area to focus on is the `models` folder, as the entire `Sequelize` setup has been converted into Typescript, but the database configuration and layout has been kept the same.
The model-specific logic functions have been nested under the corresponding modules, and are globally exported through `index.ts` as `[ModelName]Module` to be consumed through a reference to the `models/` directory (examples to such occurrences can be found in command files).

### CODE REVIEW

**Comment by Hofled** [comment]
### Let's merge this bad boy in @VuongRon 
4ujz4u

File: test/dice.test.ts [hunk 1]
```diff
@@ -108,7 +107,7 @@ describe ("Dice", () => {
     });
 
     it ("should __NOT__ detect 'a Three of a kind' before 'Full House' in [2, 5, 2, 5, 2]", () => {
-      // Full House takes precedence ofer Three of a Kind, so we have to
+      // Full House takes precedence offer Three of a Kind, so we have to
```
Comment (VuongRon) [inline test/dice.test.ts]: 👌 💯 🔥

Comment (DarkStoorM) [reply test/dice.test.ts]: Should be `over` 🏃 💨

Comment (Hofled) [reply test/dice.test.ts]: This is the highlight of the PR

Comment (Hofled) [reply test/dice.test.ts]: resolved by commit ff1e8acb518c6081ddef5303fd6ec02575d43194

**Review by dcepil — APPROVED** [review]
Looks good to me - if some things are broken we can fix them after merging this.

**Review by VuongRon — APPROVED** [review]
👴

**Pull request merged**